ui/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -283,8 +283,7 @@ export default class PermissionConnect extends Component {
         ) : (
           <PermissionConnectHeader
```

Comment:
Since we only cache the subject metadata inside this component, this fix won't work for Snaps.

I'm assuming this bug can be reproduced using the Ethereum provider example Snap

Reply:
Do you mean the title here?
https://github.com/user-attachments/assets/edb8f7a4-9fae-4426-a60a-9376a09c5795

Reply:
No, notice "Unknown Origin" here: 

https://github.com/user-attachments/assets/d7b265cd-8aaa-43e8-bbbb-9d0daeb65463